server/shared/storage/amazon.ts

```diff
@@ -0,0 +1,131 @@
+import IStorage, {
+  ContentResponse,
+  DeleteResponse,
+  ExistsResponse,
+  FileListResponse,
+  Response
+} from './interface';
+import S3, { ClientConfiguration } from 'aws-sdk/clients/s3';
+import { Config } from '../../config';
+import miss from 'mississippi';
+import path from 'path';
+import { Amazon as S3Config } from '../../config/storage';
+
+const NOT_FOUND_MESSAGE = 'Object not found';
+
+const isNotFound = (err: any): boolean => err.code === 'NoSuchKey';
+
+class Amazon implements IStorage {
+  #bucket: string;
+  #client: S3;
+
+  constructor(config: Config) {
+    const amazonConfig = config.storage.amazon as S3Config;
+
+    const s3Config = {
+      accessKeyId: amazonConfig.key,
+      secretAccessKey: amazonConfig.secret,
+      region: amazonConfig.region,
+      signatureVersion: 'v4',
+      apiVersion: '2006-03-01',
+      maxRetries: 3
+    } as ClientConfiguration;
+
+    this.#bucket = amazonConfig.bucket;
+    this.#client = new S3(s3Config);
+  }
+
+  private path(...segments): string {
+    segments = [this.#bucket, ...segments];
+    return path.join(...segments);
```

Comment:
Does this work (seems simpler)?
```suggestion
    return path.join(this.#bucket, ...segments);
```